src/figuresClasses.ts

```diff
@@ -2,10 +1,99 @@
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  color: 'red' | 'green' | 'blue';
+
+  shape: 'triangle';
+
+  a: number;
+
+  b: number;
+
+  c: number;
+
+  constructor(
+    color: 'red' | 'green' | 'blue',
+    a: number,
+    b: number,
+    c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Side must be greater than 0');
+    }
+
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error('Sides do not form a valid triangle');
+    }
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+    this.shape = 'triangle';
```

Comment:
The `shape` property is being assigned here, but it should have been initialized in the declaration itself. Change the declaration to `shape: 'triangle' = 'triangle';` and remove this line.